daemon/cli.c

```diff
@@ -199,6 +200,9 @@ static void cli_incoming_list_totals(str *instr, struct streambuf *replybuffer)
 	offer_iv = rtpe_totalstats_lastinterval.offer;
 	answer_iv = rtpe_totalstats_lastinterval.answer;
 	delete_iv = rtpe_totalstats_lastinterval.delete;
+	offers_ps = rtpe_totalstats_lastinterval.offers_ps;
+	answers_ps = rtpe_totalstats_lastinterval.answers_ps;
+	deletes_ps = rtpe_totalstats_lastinterval.deletes_ps;
```

Comment:
These aren't thread-safe, as 64-bit types aren't intrinsically atomic on all platforms, and these structs aren't protected by `rtpe_totalstats_lastinterval_lock`. Either lock each struct's mutex before copying the struct, or copy the values individually using atomic64 operations.

Reply:
`rtpe_totalstats_lastinterval` is just o copy of `rtpe_totalstats_interval` that is made once per graphite interval. When calling `send_graphite_data`, the stats are sent to graphite but they are also copied to a new struct that is later copied to `rtpe_totalstats_lastinterval`:
https://github.com/sipwise/rtpengine/blob/00bdcef9a4cd677d0c1ac36d83e4eed35917ef0c/daemon/graphite.c#L345-L353
This copy should be protected by the rtpe_totalstats_lastinterval_lock but it is not. This is an issue which I will address. 


The mutexes for each struct are used when the copy from `rtpe_totalstats_interval` is made so they don't need to be used here:
https://github.com/sipwise/rtpengine/blob/00bdcef9a4cd677d0c1ac36d83e4eed35917ef0c/daemon/graphite.c#L45-L55